src/components/HomeSection.tsx

```diff
@@ -40,14 +40,17 @@ export const HomeSection = ({
   title,
   titleDetail
 }: Props) => {
-  const {
-    data,
-    loading: isLoading,
-    refetch
-  } = useQuery(getQuery(query), {
-    variables: queryVariables,
-    fetchPolicy
-  });
+  const { data, isLoading, refetch } = useQuery(
+    [query, queryVariables],
+    async () => {
+      const client = await ReactQueryClient();
+
+      return await client.request(getQuery(query), queryVariables);
+    },
+    {
+      keepPreviousData: true
+    }
+  );
```

Comment:
How about creating a generic client provider instead? Because we're going to need it. I think creating a new provider for react-query along with react-apollo will make future integrations easier.

Reply:
yes you are right, this will need a refactoring somewhen, when we completely switch to react query. for now we keep it like this with some duplications.